internal/webhook/v1/alertmanager_config_secret_webhook.go

```diff
@@ -0,0 +1,147 @@
+	}
+	log.Info("Validation for Secret upon creation", "name", secret.GetName())
+
+	// TODO Validate tenant is in the list of accepted tenants
+
+	if err := v.validateNoDuplicateTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	return nil, validateAlertmanagerConfig(ctx, secret)
+}
+
+// validateNoDuplicateTenant ensures that no other secret with the same tenant label exists.
+func (v *AlertmanagerConfigSecretCustomValidator) validateNoDuplicateTenant(ctx context.Context, secret *corev1.Secret) error {
+	tenant, ok := secret.Labels[tenancy.TenantSelectorLabel]
+	if !ok {
+		return fmt.Errorf("tenant label is required")
+	}
+
+	var secretList corev1.SecretList
+	if err := v.client.List(ctx, &secretList, client.InNamespace(""),
+		client.MatchingLabelsSelector{
+			Selector: labels.SelectorFromSet(
+				labels.Set{
+					predicates.AlertmanagerConfigSelectorLabelName: predicates.AlertmanagerConfigSelectorLabelValue,
+					tenancy.TenantSelectorLabel:                    tenant,
+				},
+			),
+		}); err != nil {
+		return fmt.Errorf("failed to list secrets for tenant %s: %w", tenant, err)
+	}
+
+	if len(secretList.Items) > 0 {
+		return fmt.Errorf("a secret for tenant %s already exists", tenant)
+	}
+	return nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type Secret.
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	secret, ok := newObj.(*corev1.Secret)
+	if !ok {
+		return nil, fmt.Errorf("expected a Secret object for the newObj but got %T", newObj)
+	}
+	log.Info("Validation for Secret upon update", "name", secret.GetName())
+
+	// TODO Validate tenant is in the list of accepted tenants
+
+	// TODO check for duplicates if the secret is being updated with a different tenant
+
+	return nil, validateAlertmanagerConfig(ctx, secret)
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type Secret.
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
+	}
+	log.Info("Validation for Secret upon deletion", "name", secret.GetName())
+
+	// We have nothing to validate on deletion
+	return nil, nil
+}
+
+func validateAlertmanagerConfig(ctx context.Context, secret *corev1.Secret) error {
+	// Check that the secret contains an "alertmanager.yaml" file.
+	alertmanagerConfig, found := secret.Data["alertmanager.yaml"]
+	if !found {
+		return fmt.Errorf("missing alertmanager.yaml in the secret")
+	}
+	cfg, err := config.Load(string(alertmanagerConfig))
+	if err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+	// TODO add more validation on the templates directly
+	log.Info("alertmanager config validation successful", "config", cfg)
```

Comment:
We should not print the config here, it might contain sensitive information like api keys/tokens and various other informations.